dataland-external-storage/src/main/kotlin/org/dataland/datalandexternalstorage/services/EurodatDataStore.kt

```diff
@@ -0,0 +1,275 @@
+import org.springframework.messaging.handler.annotation.Payload
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.sql.SQLException
+
+/**
+ * Simple implementation of a data storing service using the EuroDaT data trustee
+ * @param cloudEventMessageHandler service for managing CloudEvents messages
+ * @param temporarilyCachedDataClient the service for retrieving data from the temporary storage
+ * @param temporarilyCachedDocumentClient the service for retrieven documents from the temporary storage
+ * @param objectMapper object mapper used for converting data classes to strings and vice versa
+ * @param databaseCredentialResourceClient the service to retrieve eurodat storage credentials
+ * @param safeDepositDatabaseResourceClient the service to create the safe deposit box used to store private data
+ * on eurodat
+ * @param messageUtils contains utils connected to the messages on the message queue
+ */
+@Component
+class EurodatDataStore(
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired var temporarilyCachedDataClient: TemporarilyCachedDataControllerApi,
+    @Autowired var temporarilyCachedDocumentClient: StreamingTemporarilyCachedPrivateDocumentControllerApi,
+    @Autowired var databaseCredentialResourceClient: DatabaseCredentialResourceApi,
+    @Autowired var safeDepositDatabaseResourceClient: SafeDepositDatabaseResourceApi,
+    @Autowired var objectMapper: ObjectMapper,
+    @Autowired var messageUtils: MessageQueueUtils,
+    @Value("\${dataland.eurodatclient.app-name}")
+    private val eurodatAppName: String,
+    @Value("\${dataland.eurodatclient.max-retries-connecting}")
+    private val maxRetriesConnectingToEurodat: Int,
+    @Value("\${dataland.eurodatclient.milliseconds-between-retries}")
+    private val millisecondsBetweenRetriesConnectingToEurodat: Int,
+    @Value("\${dataland.eurodatclient.initialize-safe-deposit-box}")
+    private val initializeSafeDepositBox: Boolean,
+) {
+    // TODO try to move the variables here
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Tries to create a safe deposit box in EuroDaT for storage of Dataland data a pre-defined number of times and
+     * then throws a final exception after the retries are used up.
+     */
+    @PostConstruct
+    fun createSafeDepositBox() {
+        if (initializeSafeDepositBox) {
+            logger.info("Checking if safe deposit box exits. If not creating safe deposit box")
+            retryWrapperMethod("createSafeDepositBox") {
+                isSafeDepositBoxAvailable()
+            }
+        }
+    }
+
+    /**
+     * This method will rerun a given method if an exception is thrown while running it
+     * @param inputMethod to specify in the logs which method should be rerun
+     */
+    @Suppress("TooGenericExceptionCaught")
```

Comment:
Confirm that in these cases here it is fine to useSuppress("TooGenericExceptionCaught") and Suppress("TooGenericExceptionThrown") as they are only caught to trigger the rerun logic and if they persit are thrown after the last retry has failed